transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemAdd.java

```diff
@@ -473,6 +474,14 @@ private void performRecoveryEnvBoottime(OperationContext context, ModelNode mode
                 .addService(TxnServices.JBOSS_TXN_EXTENDED_JBOSS_XA_TERMINATOR, extendedJBossXATerminatorService)
                 .setInitialMode(Mode.ACTIVE).install();
 
+        final JBossContextXATerminatorService contextXATerminatorService = new JBossContextXATerminatorService();
+        context.getServiceTarget()
+                .addService(TxnServices.JBOSS_TXN_CONTEXT_XA_TERMINATOR, contextXATerminatorService)
```

Comment:
iiuc it seems OK, I was concerned about services not starting in time due to the dependency order (the Extended terminator, JBoss terminator and transaction service) but if after your investigation you believe the ordering is right to get those alive at the right time I am OK